src/test/scala/api/API.scala

```diff
@@ -148,6 +150,7 @@ class APIIsBackwardCompatible extends org.scalatest.FeatureSpec {
       import com.microsoft.azure.iot.iothubreact.checkpointing.backends.CheckpointBackend
 
       class CustomBackend extends CheckpointBackend {
+        val cfg = new Configuration()
```

Comment:
see my comment about ICPConfiguration. I would add `MochitoSugar` and mock ICPConfiguration instead (also considering that the Configuration class should remain private